test/e2e/validators.go

```diff
@@ -18,251 +20,228 @@ import (
 
 	buildv1alpha1 "github.com/redhat-developer/build/pkg/apis/build/v1alpha1"
 	taskv1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/kubectl/pkg/scheme"
 )
 
 var (
-	EnvVarImageRepo       = "TEST_IMAGE_REPO"
-	EnvVarImageRepoSecret = "TEST_IMAGE_REPO_SECRET"
-	EnvVarSourceURLGithub = "TEST_PRIVATE_GITHUB"
-	EnvVarSourceURLGitlab = "TEST_PRIVATE_GITLAB"
-	EnvVarSourURLSecret   = "TEST_SOURCE_SECRET"
-	QuayHostURL           = "quay.io"
+	EnvVarImageRepo            = "TEST_IMAGE_REPO"
+	EnvVarEnablePrivateRepos   = "TEST_PRIVATE_REPO"
+	EnvVarImageRepoSecret      = "TEST_IMAGE_REPO_SECRET"
+	EnvVarSourceRepoSecretJSON = "TEST_IMAGE_REPO_DOCKERCONFIGJSON"
+	EnvVarSourceURLGithub      = "TEST_PRIVATE_GITHUB"
+	EnvVarSourceURLGitlab      = "TEST_PRIVATE_GITLAB"
+	EnvVarSourceURLSecret      = "TEST_SOURCE_SECRET"
 )
 
-// OperatorEmulation is used as an struct
-// to hold required data
-type OperatorEmulation struct {
-	buildRun                *operator.BuildRun
-	clusterBuildStrategy    *operator.ClusterBuildStrategy
-	buildStrategy           *operator.BuildStrategy
-	build                   *operator.Build
-	namespace               string
-	identifier              string
-	buildStrategySamplePath string
-	buildSamplePath         string
-	buildRunSamplePath      string
+// cleanupOptions return a CleanupOptions instance.
+func cleanupOptions(ctx *framework.TestCtx) *framework.CleanupOptions {
+	return &framework.CleanupOptions{
+		TestContext:   ctx,
+		Timeout:       cleanupTimeout,
+		RetryInterval: cleanupRetryInterval,
+	}
 }
 
-func newOperatorEmulation(n string, id string, bSPath string, bPath string, bRPath string) *OperatorEmulation {
-	return &OperatorEmulation{
-		buildRun:                &operator.BuildRun{},
-		clusterBuildStrategy:    &operator.ClusterBuildStrategy{},
-		buildStrategy:           &operator.BuildStrategy{},
-		build:                   &operator.Build{},
-		namespace:               n,
-		identifier:              id,
-		buildStrategySamplePath: bSPath,
-		buildSamplePath:         bPath,
-		buildRunSamplePath:      bRPath,
+// createContainerRegistrySecret use environment variables to check for container registry
+// credentials secret, when not found a new secret is created.
+func createContainerRegistrySecret(t *testing.T, ctx *framework.TestCtx, f *framework.Framework) {
+	secretName := os.Getenv(EnvVarImageRepoSecret)
+	secretPayload := os.Getenv(EnvVarSourceRepoSecretJSON)
+	if secretName == "" && secretPayload == "" {
```

Comment:
Does this mean that it is mandatory to provided a secret name and/or payload?

Reply:
Nope, it means it's optional. When both variables are empty, the container registry secret is not created.

Reply:
Got it!